plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFileOperations.java

```diff
@@ -0,0 +1,189 @@
+                        .add(new FileOperation("FileSystem.listFiles", "test_cache_file_operations"))
+                        .add(new FileOperation("FileSystem.listFiles", "key=p1"))
+                        .build());
+        assertFileSystemAccesses(
+                "SELECT * FROM test_cache_file_operations",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation("Input.readTail", "key=p1/"))
+                        .add(new FileOperation("FileSystem.listFiles", "test_cache_file_operations"))
+                        .build());
+
+        assertUpdate("INSERT INTO test_cache_file_operations VALUES ('2-xyz', 'p2')", 1);
+        assertFileSystemAccesses(
+                "SELECT * FROM test_cache_file_operations",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation("Input.readTail", "key=p1/"))
+                        .add(new FileOperation("Input.readTail", "key=p2/"))
+                        .add(new FileOperation("FileSystem.listFiles", "test_cache_file_operations"))
+                        .add(new FileOperation("FileSystem.listFiles", "key=p2"))
+                        .build());
+        assertFileSystemAccesses(
+                "SELECT * FROM test_cache_file_operations",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation("Input.readTail", "key=p1/"))
+                        .add(new FileOperation("Input.readTail", "key=p2/"))
+                        .add(new FileOperation("FileSystem.listFiles", "test_cache_file_operations"))
+                        .build());
+
+        assertQuerySucceeds("CALL system.flush_metadata_cache()");
+        assertFileSystemAccesses(
+                "SELECT * FROM test_cache_file_operations",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation("Input.readTail", "key=p1/"))
+                        .add(new FileOperation("Input.readTail", "key=p2/"))
+                        .add(new FileOperation("FileSystem.listFiles", "test_cache_file_operations"))
+                        .add(new FileOperation("FileSystem.listFiles", "key=p1"))
+                        .add(new FileOperation("FileSystem.listFiles", "key=p2"))
+                        .build());
+        assertFileSystemAccesses(
+                "SELECT * FROM test_cache_file_operations",
+                ImmutableMultiset.<FileOperation>builder()
+                        .add(new FileOperation("Input.readTail", "key=p1/"))
+                        .add(new FileOperation("Input.readTail", "key=p2/"))
+                        .add(new FileOperation("FileSystem.listFiles", "test_cache_file_operations"))
+                        .build());
+
+        assertQuerySucceeds("CALL system.flush_metadata_cache(schema_name => CURRENT_SCHEMA, table_name => 'test_cache_file_operations')");
```

Comment:
Let's add another assertion which verifies that flushing the cache for a different table, does not affect the directory caching for `test_cache_file_operations`

Reply:
+1 ^, we could have 2nd table from beginning where `flush_metadata_cache` (without parameters) verify against all the tables.